Source/utils/file_util.cpp

```diff
@@ -39,6 +39,20 @@ bool FileExists(const char *path)
 #endif
 }
 
+bool IsWriteable(const char *path)
+{
+	struct stat stats;
+	if (stat(path, &stats) == 0) {
+		if (stats.st_mode & 2) {
+			return true;
+		} else {
+			return false;
+		}
+	} else {
+		return false;
+	}

```

Comment:
Some useless else and nested ifs:
```suggestion
	if (stat(path, &stats) == 0 && stats.st_mode & 2) {
		return true;
	}
	
	return false;
```